src/foam/nanos/dig/DigWebAgent.java

```diff
@@ -209,7 +209,14 @@ public void put(Object o, Detachable d) {
 
       FObject   obj      = null;
       ClassInfo cInfo    = dao.getOf();
-      Class     objClass = cInfo.getObjClass();
+      Class     objClass = null;
+
+      if ( cInfo != null ) {
+        objClass = cInfo.getObjClass();
+      } else {
+         out.println("Can't find DAO");
```

Comment:
Is this correct? If you didn't find the dao it would be null on line 211 and would throw a NullPointerException.

Reply:
Don't worry about this, I've completely re-written this.